src/marks/auto.js

```diff
@@ -90,7 +90,18 @@ export function autoSpec(data, options) {
       colorMode = "stroke";
       break;
     case "line":
-      markImpl = X && Y ? line : X ? lineX : lineY; // 1d line by index
+      markImpl =
+        X && Y // same logic as area (see below), but default to line
+          ? yZero
+            ? lineY
+            : xZero || isMonotonic(Y)
+            ? lineX
+            : isMonotonic(X)
+            ? lineY
+            : line
+          : X // 1d line by index
+          ? lineX
+          : lineY;
```

Comment:
If both **x** and **y** are monotonic, and neither **xZero** nor **yZero** are specified, I think we should default to lineY rather than lineX. If this works, I think I’ll push this:

```suggestion
      markImpl =
        X && Y // same logic as area (see below), but default to line
          ? yZero || isMonotonic(X)
            ? lineY
            : xZero || isMonotonic(Y)
            ? lineX
            : line
          : X // 1d line by index
          ? lineX
          : lineY;
```